guides/common/modules/proc_configuring-a-puppet-class.adoc

```diff
@@ -1,5 +1,6 @@
 [id="puppet_guide_configuring_a_puppet_class_{context}"]
 = Configuring a Puppet Class
+// = Overriding Parameters in a Puppet Class ??? Globally?
```

Comment:
Yes. Puppet modules have built-in defaults but within the application you can override those defaults. If you want to override a parameter within the application (globally, on a host group or on hosts), it must be marked as overrideable.

One thing this doesn't mention (perhaps it's mentioned elsewhere) is that you can also mark a parameter as overrideable but then not override the built-in default. There is a checkbox `Omit` to do so. IMHO that should be the default, but the documentation should at the very least mention it.

Reply:
@ekohl I can add it, but I would prefer to know in which use cases is benefical to only mark a parameter overridable without overriding its value.

Reply:
If you want to use the built in default from the Puppet manifest. Let's say you have:
```puppet
class myclass (
  String $servername = $facts['fqdn'],
) {
  # ...
}
```

This essentially means you have a class named `myclass`. If you import it, it will find the parameter `servername` with a default `$facts['fqdn']`. If you click on `override`, that literal string is copied into Foreman's default field.

To really get an equivalent default, you would need to use some ERB macro that I don't even know from the top of my head. Something like `<%= ... %>` sould be the default in Foreman's default field. If you use `omit`, you can continue using the built-in Puppet default.

That is to me, a Puppet user, a way more natural way. Only if I want to override the built-in default with a new global default should I provide a default.

Also note that leaving the default field empty is AFAIK the empty string. Selecting `Omit` is fundamentally different.

If you have an environment where you can play, I'd recommend to try it out. If you go to the host detail page you should see a button YAML (though I think it's renamed when it was extract to a plugin). There you should see the assigned class with overrides. With `Omit`, it should not show up unless overridden at the host level. Without it will always show up.

Reply:
@ekohl You know, this is a great remark but it's not what this PR is concerned with. So I suggest that I will create an issue in our Jira for future work, because we will get to some rewrites of the content later. Agreed?
With this comment I was merely wondering if a different phrasing of the heading would be more suitable.

Reply:
That's completely OK, but at least you now have a bit more understanding of the background.

Reply:
Absolutely and I appreciate it!